cmake/CallbackSetupExtraOptions.cmake

```diff
@@ -92,3 +92,6 @@ include("${Trilinos_SOURCE_DIR}/commonTools/build_stats/BuildStatsWrappers.cmake
 generate_build_stats_wrappers()
 remove_build_stats_file_on_configure()
 remove_build_stats_timing_files_on_fresh_configure()
+
+# Set up C++ language standard selection
+include(SetTrilinosCxxStandard)
```

Comment:
This should do it but I am not sure why stuff got placed outside of the macro `TRIBITS_REPOSITORY_SETUP_EXTRA_OPTIONS()` that gets defined in this file and called by TriBITS.  Semantically, there is not a big difference in behavior except that this code outside of the macro `TRIBITS_REPOSITORY_SETUP_EXTRA_OPTIONS()` in that file actually gets called before the code inside of the macro as you can see in:

https://github.com/trilinos/Trilinos/blob/3ff154a7d724400b53afa726d868408ede667e39/cmake/tribits/core/package_arch/TribitsPackagingSupport.cmake#L312-L314

But I see that I was the one who set the precedent for this by adding the build_stats stuff there in commit 3ea94177a4a38e847fb5a78fe0d9edf7d7ebd1d4 with no explanation for why that was done.

SIDENOTE: I don't remember why I decided to put things in a macro in these files because just including CMake code in a file is largely equivalent to calling a macro.

Reply:
I'll do whichever you think is best.  I noticed that too, but decided to put it in (IMO) the simpler spot

Reply:
We can do this in a separate PR now that PR testing has already started.